MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -7240,5 +7254,110 @@ private void strlwr()
+        }
+
+        /// <summary>
+        ///     Generic "Has Key" routine which takes in the USER struct vs. the current channel
+        ///
+        ///     Signature: int ok=gen_haskey(char *lock, int unum, struct user *uptr);
+        ///     Returns: AX = 1 == True
+        /// </summary>
+        /// <returns></returns>
+        private void gen_haskey()
+        {
+            var lockNamePointer = GetParameterPointer(0);
+            var lockNameBytes = Module.Memory.GetString(lockNamePointer, true);
+
+#if DEBUG
+            _logger.Info($"Returning TRUE for Haskey({Encoding.ASCII.GetString(lockNameBytes)})");
+#endif
+            Registers.AX = 1;
+        }
+
+        /// <summary>
+        ///     Expect a Word from the user (character from the current command)
+        ///
+        ///     cncwrd() is executed after begincnc(), which runs rstrin() replacing the null separtors
+        ///     in the string with spaces once again.
+        /// </summary>
+        private void cncwrd()
+        {
+            //Get Input
+            var inputPointer = Module.Memory.GetVariablePointer("INPUT");
+            var nxtcmdPointer = Module.Memory.GetPointer("NXTCMD");
+            var inputLength = Module.Memory.GetWord("INPLEN");
+
+            var remainingCharactersInCommand = inputLength - (nxtcmdPointer.Offset - inputPointer.Offset);
+
+            //Skip any excessive spacing
+            while (Module.Memory.GetByte(nxtcmdPointer) == ' ' && remainingCharactersInCommand > 0)
+            {
+                nxtcmdPointer.Offset++;
+                remainingCharactersInCommand--;
+            }
+            var returnPointer = Module.Memory.GetOrAllocateVariablePointer("CNCWORD", 0x1E); //max length is 30 characters
+            Registers.DX = returnPointer.Segment;
+            Registers.AX = returnPointer.Offset;
+
+            //Verify we're not at the end of the input
+            if (remainingCharactersInCommand == 0 || Module.Memory.GetByte(nxtcmdPointer) == 0)
+            {
+#if DEBUG
+                _logger.Info($"End of Input");
+#endif
+                //Write null to output
+                Module.Memory.SetByte(returnPointer, 0);
+                return;
+            }
+
+            var returnedWord = new MemoryStream();
+            var inputString = Module.Memory.GetArray(nxtcmdPointer, (ushort)remainingCharactersInCommand);
+            
+            //Build Return Word stopping when a space is encountered
+            foreach (var b in inputString)
+            {
+                if (b == ' ' || b == 0)
+                    break;
+
+                returnedWord.WriteByte(b);
+            }
+            
+            //Truncate to 29 bytes
+            if(returnedWord.Length > 29)
+                returnedWord.SetLength(29);
+
+            returnedWord.WriteByte(0);
+            
+            Module.Memory.SetArray(returnPointer, returnedWord.ToArray());
+
+#if DEBUG
```

Comment:
With exhaustive unit tests, do we need the log spam?

Reply:
Probably not -- they're usually more in there as a tracer on when things would go wrong earlier on with Memory management. But you're right, probably don't need the obvious so verbose now 👍